tests/unit/Extension/ExtensionCollectionTest.php

```diff
@@ -0,0 +1,114 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OAT\DependencyResolver\Extension;
+
+use OAT\DependencyResolver\Extension\Entity\Extension;
+use PHPUnit\Framework\TestCase;
+
+class ExtensionCollectionTest extends TestCase
+{
+    /**
+     * @var ExtensionCollection
+     */
+    private $subject;
+
+    public function setUp()
+    {
+        $this->subject = new ExtensionCollection();
+    }
+
+    public function testConstructorReturnsAnEmptyCollection()
+    {
+        $this->assertInstanceOf(ExtensionCollection::class, $this->subject);
+        $this->assertCount(0, $this->subject->getIterator());
+    }
+
+    public function testAddWithValidExtensionReturnsCollectionWithOneExtension()
+    {
+        $extensionName = 'extensionName';
+        /** @var Extension $extension */
+        $extension = $this->createConfiguredMock(Extension::class, ['getExtensionName' => $extensionName]);
+
+        $this->assertInstanceOf(ExtensionCollection::class, $this->subject->add($extension));
+
+        $this->assertCount(1, $this->subject->getIterator());
+        $this->assertTrue($this->subject->offsetExists($extensionName));
+        $this->assertEquals($extension, $this->subject->offsetGet($extensionName));
+    }
+
+    public function testOffsetSetWithInvalidExtensionThrowsException()
+    {
+        $extensionName = 'extensionName';
+        $extension = 'not an Extension object';
+        $this->expectException(\TypeError::class);
+        $this->subject->offsetSet($extensionName, $extension);
+    }
+
+    public function testOffsetGetWithNotExistingExtensionReturnsNull()
+    {
+        $extensionName = 'extensionName';
+        /** @var Extension $extension */
+        $this->createConfiguredMock(Extension::class, ['getExtensionName' => $extensionName]);
+        $this->assertNull($this->subject->offsetget($extensionName . 'blah'));
```

Comment:
Please fix typo in method name (case mismatch)

Reply:
Done